selfdrive/car/gm/carcontroller.py

```diff
@@ -133,12 +130,18 @@ def update(self, sendcan, enabled, CS, frame, actuators, \
       if (frame % 4) == 0:
         idx = (frame / 4) % 4
 
-        at_full_stop = enabled and CS.standstill
-        near_stop = enabled and (CS.v_ego < P.NEAR_STOP_BRAKE_PHASE)
+        car_stopping = apply_gas < P.ZERO_GAS
```

Comment:
what happens if you are stopping uphill and apply_gas remains positive?

Reply:
ah, didn't think of that (didn't test it on very steep hills)

Reply:
This is a fundamental issue with openpilot's interface design. The stopping intent is never passed up to an interface.

Reply:
Another problem is that GM interface doesn't know if car is moving forward or backwards.